Buld fails with clang / possible memleak

```
$ make
clang -o tls_example tls_example.cpp -std=c++17 -fpic -Wall -Werror -Wextra -pedantic -lpthread -lstdc++ -lssl -lcrypto
[...]
/usr/bin/../lib/gcc/i686-linux-gnu/8/../../../../include/c++/8/optional:200:11: error: destructor called on
      non-final 'net::tls_connection<net::ip_version::v4>' that has virtual functions but non-virtual destructor
[...]
1 error generated.
make: *** [Makefile:14: tls] Fehler 1
```
tells that the base class constructor is not virtual which might cause memleaks.

Easy to fix with
```
diff --git a/include/socketwrapper/detail/base_socket.hpp b/include/socketwrapper/detail/base_socket.hpp
index 230cde9..f575bd3 100644
--- a/include/socketwrapper/detail/base_socket.hpp
+++ b/include/socketwrapper/detail/base_socket.hpp
@@ -57,7 +57,7 @@ public:
 #endif
     }
 
-    ~base_socket()
+    virtual ~base_socket()
     {
         if(m_sockfd > 0)
         {
```

To build using clang I used
```
diff --git a/example/Makefile b/example/Makefile
index cf6996a..56322d5 100644
--- a/example/Makefile
+++ b/example/Makefile
@@ -1,7 +1,7 @@
-cc = g++
+cc = clang
 
 CFLAGS = -std=c++17 -fpic -Wall -Werror -Wextra -pedantic
-LDFLAGS = -lpthread
+LDFLAGS = -lpthread -lstdc++
 
 %.o: %.cpp
        $(cc) -c $< -o $@ $(CFLAGS)
```

--- a/example/Makefile
+++ b/example/Makefile
@@ -1,4 +1,5 @@
-cc = g++
+#cc = g++
+cc = clang++
 
 CFLAGS = -std=c++17 -fpic -Wall -Werror -Wextra -pedantic
 LDFLAGS = -lpthread

--- a/include/socketwrapper/detail/base_socket.hpp
+++ b/include/socketwrapper/detail/base_socket.hpp
@@ -57,7 +57,7 @@ public:
 #endif
     }
 
-    ~base_socket()
+    virtual ~base_socket()
     {
         if(m_sockfd > 0)
         {

